includes/Utilities/Plugin_Request_Utility.php

```diff
@@ -181,4 +181,63 @@ public static function get_files_to_ignore() {
 
 		return $files_to_ignore;
 	}
+
+	/**
+	 * Downloads the plugin from the URL.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param string $plugin_url The URL of the plugin to download.
+	 * @return string|bool The plugin basename if the plugin is downloaded successfully, false otherwise.
+	 */
+	public static function download_plugin( $plugin_url ) {
+		$response = wp_remote_get( $plugin_url );
```

Comment:
What happen if someone add zip that is not plugin? For example theme zip?

Reply:
Hmm, I have not checked. This PR uses https://github.com/WordPress/plugin-check/pull/490 arbitrary path approach. I will create follow-up issue for that PR.

Reply:
Issue - https://github.com/WordPress/plugin-check/issues/547